core/arch/arm/mm/tee_mmu.c

```diff
@@ -263,7 +263,7 @@ TEE_Result tee_mmu_init(struct tee_ta_ctx *ctx)
 TEE_Result tee_mmu_map(struct tee_ta_ctx *ctx, struct tee_ta_param *param)
 {
 	TEE_Result res = TEE_SUCCESS;
-	paddr_t pa;
+	paddr_t pa = NULL;
```

Comment:
I don't see the connection between physical address 0 and `-fno-delete-null-pointer-checks`. For virtual addresses we can always make sure to have address 0 unmapped, but we can't control the physical ones.

If we only may pass valid pointers to `core_va2pa()` then it can never fail and we don't need a status return value. In principle I see no problem changing the prototype.